### PR TITLE
Arduino ultrasonic sensor project

### PR DESCRIPTION
**Pull request by iainardo**
Hi folks, I've added a new electronic project post for review. Project deals with using an ultrasonic sensor and an arduino as a proximity meter. Let me know what you think. Thanks

### CODE REVIEW

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
+---
+layout:       article
+lang:         en
+parent:       electronic-resources
+breadcrumb:   true
+permalink:    /en/projects/electronic/ultrasonic-sensor-2019
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: should be `/en/resources/electronic-resources/ultrasonic-sensor-2019` to be consistent with the sibling pages

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+## Ardunino Ultrasonic range sensor 
+
+This project will use an ultrasonic range sensor as a proximity meter.
+
+<img class="img-fluid" src="{{'assets/posts/2019-09-20-arduino-ultrasonic-sensor/ultrasonic-sensor.png' | relative_url}}"/>
+
+
+### Components
+THe following components are required for this project
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: TH typo

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+* Arduino Uno R3
+* HC-SR04 Ultrasonic Sensor 
+* 1 kΩ Resistor (x 3)
+* Green LED
+* Yellow LED
+* Red LED3
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: *LED3* typo?

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+* Piezo buzzer
+
+### About this project
+
+This project will use a ultrasonic sensor to detect motion and light different leds and sound a buzzer
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: use a**n** ultrasonic (typo)

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+depending on the distance of an object from the sensor:
+  
+We will use Green, Yellow, Red leds and buzzer to signal distance of object as follows: 
+ * Light the GREEN LED if the object is greater than 100cm away from sensor
+ * Light the YELLOW LED if the object is btween 70cm and  100cm away from sensor
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: btween (typo)

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+permalink:    /en/projects/electronic/ultrasonic-sensor-2019
+ref:          2020-ultrasonic-sensor-arduino
+title:        Ultrasonic sensor with Arduino
+author:       Iain Campbell
+description:  How to use an ultrasonic sensor to detect motion
+tags:         Arduino, electronic, ultrasonic sensor, led, buzzer
+code:         true
+---
+
+## Ardunino Ultrasonic range sensor
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: Ardunino (typo)

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
+---
+layout:       article
+lang:         en
+parent:       electronic-resources
+breadcrumb:   true
+permalink:    /en/projects/electronic/ultrasonic-sensor-2019
+ref:          2020-ultrasonic-sensor-arduino
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: 2020 You're in advance on the calendar ;)

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+   For example, if the object is 10 cm away from the sensor, and the speed of the sound is 343 m/s or 0.0343 cm/µs the sound wave will need to travel about 294 µ seconds. 
+   But what you will get from the Echo pin will be double that number because the sound wave needs to travel forward and bounce backward.  So in order to get the distance in cm we need to multiply the received travel time value from the echo pin by 0.0343 and divide it by 2.
+
+
+#### HC-SR04 Timing diagram
+
+* The circuit needs to supply a short 10uS pulse to the trigger input to start the ranging, and then the module will send out
+an 8 cycle burst of ultrasound at 40 kHz and raise its echo. 
+* The Echo is avdistance object that is pulse width and the range in proportion .You can
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: - avdistance (typo)
- I'm confused by the sentence, can you check it out?

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+range = high level time * velocity (340M/S) / 2; we suggest to use over 60ms
+measurement cycle, in order to prevent trigger signal to the echo signal. 
+
+<img class="img-fluid" src="{{'assets/posts/2019-09-20-arduino-ultrasonic-sensor/ultrasonic-timing-diagram.png' | relative_url}}"/>
+
+
+
+### Component diagram
+
+The following shows the a breadboard component diagram which will be used for this project.
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: shows the a breadboard

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+
+<img class="img-fluid" src="{{'assets/posts/2019-09-20-arduino-ultrasonic-sensor/ultrasonic-range-sensor.png' | relative_url}}"/>
+
+
+### Code
+
+* The following code snippet can be used  (note it's partially incomplete - you can complete yourself)
+
+```
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: The syntax highlighter supports `cpp`.
See https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+void yellow(){
+  // light yellow..
+}
+
+```
+
+
+### Resources
+* Lookup the data sheet for the HC-SR04 to find more out about its specifications and how to operate.
+* Prior to building the circuit for real, you could use [Tinkercad](http://tinkercad.com) to prototype
```
Comment (marc-ed-raffalli) [inline _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: *you could use*, I would put a stronger statement like: *We strongly recommend...*

**Review by marc-ed-raffalli — COMMENTED** [review]
It demystifies very well the principles of a sonar great stuff, nicely illustrated as well!

Few minor typos and the `permalink` to be updated.

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md (hunk 7 above)
Comment (iainardo) [reply _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: Thanks! Sorry, I'm time travelling again

File: _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md (hunk 8 above)
Comment (iainardo) [reply _posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md]: Thanks. Yes, it's confusing..I've reworded..

**Review by marc-ed-raffalli — APPROVED** [review]

**Pull request merged**